sunpy/map/mapbase.py

```diff
@@ -2158,19 +2161,26 @@ def draw_quadrangle(self, bottom_left, *, width: u.deg = None, height: u.deg = N
         """
         axes = self._check_axes(axes, allow_non_wcsaxes=False)
 
-        bottom_left, top_right = get_rectangle_coordinates(
-            bottom_left, top_right=top_right, width=width, height=height)
+        if isinstance(bottom_left, astropy.units.quantity.Quantity):
+            anchor, _, top_right, _ = self._parse_submap_quantity_input(bottom_left, top_right, width, height)
+            width, height = top_right - anchor
+            kwargs.update({"vertex_unit": u.pix})
```

Comment:
```suggestion
            transform = axes.get_transform(self.wcs)
            kwargs.update({"vertex_unit": u.pix})
```
This allows for plotting a rectangle in the pixel space of the map onto axes that may correspond to a different WCS.